lib/puppet_x/templates/invoke_ps_command.erb

```diff
@@ -151,38 +395,53 @@ try
 
   $ps = [System.Management.Automation.PowerShell]::Create()
   $ps.Runspace = $runspace
-
   [Void]$ps.AddScript($ourFunctions)
   $ps.Invoke()
 
+
   if(!$environmentVariables){
+    $ps.Commands.Clear()
     $environmentVariables = $ps.AddCommand('Get-ProcessEnvironmentVariables').Invoke()
   }
 
   if($PSVersionTable.PSVersion -le [Version]'2.0'){
     if(!$psVariables){
       $psVariables = $ps.AddScript('Get-Variable').Invoke()
     }
+
+    $ps.Commands.Clear()
     [void]$ps.AddScript('Get-Variable -Scope Global | Remove-Variable -Force -ErrorAction SilentlyContinue -WarningAction SilentlyContinue')
     $ps.Invoke()
+
+    $ps.Commands.Clear()
     [void]$ps.AddCommand('Reset-ProcessPowerShellVariables').AddParameter('psVariables', $psVariables)
     $ps.Invoke()
   }
 
+  $ps.Commands.Clear()
   [Void]$ps.AddCommand('Reset-ProcessEnvironmentVariables').AddParameter('processVars', $environmentVariables)
   $ps.Invoke()
 
+  $ps.Commands.Clear()
   [Void]$ps.AddScript($powershell_code)
+
+  # out-default and MergeMyResults takes all output streams
+  # and writes it to the PSHost we create
+  # this needs to be the last thing executed
+  [void]$ps.AddCommand("out-default");
+  $ps.Commands.Commands[0].MergeMyResults([System.Management.Automation.Runspaces.PipelineResultTypes]::All, [System.Management.Automation.Runspaces.PipelineResultTypes]::Output);
   $asyncResult = $ps.BeginInvoke()
 
   if (!$asyncResult.AsyncWaitHandle.WaitOne(<%= timeout_ms %>, $false)){
     throw "Catastrophic failure: PowerShell module timeout (<%= timeout_ms %> ms) exceeded while executing"
   }
 
   $output = $ps.EndInvoke($asyncResult)
```

Comment:
What's in `$output` now?  The variable is no longer used after calling `EndInvoke` now...